pages/1.10/security/ent/tls-ssl/ca-custom/index.md

```diff
@@ -8,26 +8,306 @@ excerpt:
+- **Configuration:** The set of the configuration parameters that governs the specific aspects of the installation procedure. The configuration is stored in the DC/OS configuration file.
+
+- **DC/OS configuration file:** The file which contains the DC/OS configuration parameters. The DC/OS configuration file is normally called `config.yaml` and must be present in the `genconf/` directory on the bootstrap node during the installation. It is used by the DC/OS installer.
+
+- **Installation directory:** The directory on the bootstrap node where the DC/OS installer resides. It is denoted with `$DCOS_INSTALL_DIR` in this document.
+
+# Requirements
+
+In order to install DC/OS Enterprise with a custom CA certificate you will need:
+
+- to use the **Advanced installation method:**  [advanced DC/OS installation method](/1.10/installing/custom/advanced/). Other installation methods are not supported.
+- **A file containing the custom CA certificate**
+- **A file containing the private key associated with the custom CA certificate**
+- If the CA is **not** a root CA, **a file containing the certificate chain associated with the custom CA certificate** 
+
+## Manually placing the custom CA certificate, the associated private key and the certificate chain onto the bootstrap node
+The custom CA certificate, the associated private key and the certificate chain files must be put in the `$DCOS_INSTALL_DIR/genconf/` directory on the bootstrap node:
+
+```bash
+cd $DCOS_INSTALL_DIR
+ls genconf/
+```
+```bash 
+dcos-ca-certificate.crt
+dcos-ca-certificate-key.key
+dcos-ca-certificate-chain.crt
+```
+
+## Manually placing the private key associated with the custom CA certificate onto the master nodes
+
+For security reasons, the installer will not copy the private key from the bootstrap node to the master nodes. 
+The private key associated with the custom CA certificate must be distributed manually to every DC/OS master node **before starting the installation**. The filesystem path for the private key file must be `/var/lib/dcos/pki/tls/CA/private/custom_ca.key`. The directory `/var/lib/dcos/pki/tls/CA/private` should be created manually with the following  command before putting the file `custom_ca.key` in the directory on every DC/OS master node:
+
+```bash
+ mkdir -p /var/lib/dcos/pki/tls/CA/private
+```
+
+Furthermore, the file must be owned by the root Unix user and to have 0600 permissions set.
+
+If you copy private key file over the network onto the master nodes, the network channel must be adequately protected.
+
+An example of copying the CA private key is given below. The commands are executed on the bootstrap node. The `W.X.Y.Z` denotes the IP address of a master node below.
+
+```bash
+cd $DCOS_INSTALL_DIR/config
+scp custom_ca.key centos@W.X.Y.Z:/var/lib/dcos/pki/tls/CA/private
```

Comment:
To make the examples follow the flow, please make this command use the same filename as the bootstrap node above, i.e.:
```
scp dcos-ca-certificate-key.key centos@W.X.Y.Z:/var/lib/dcos/pki/tls/CA/private/custom_ca.key
```

If the installer copies the key from the bootstrap node, then this command will work properly, and even if they copy from another machine, they probably have used the same filename as for the bootstrap node.